apis/v1alpha2/tlsroute_types.go

```diff
@@ -46,17 +46,44 @@ type TLSRoute struct {
 
 // TLSRouteSpec defines the desired state of a TLSRoute resource.
 type TLSRouteSpec struct {
-	// Rules are a list of TLS matchers and actions.
-	//
-	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=16
-	Rules []TLSRouteRule `json:"rules"`
-
 	// Gateways defines which Gateways can use this Route.
 	//
 	// +optional
 	// +kubebuilder:default={allow: "SameNamespace"}
 	Gateways *RouteGateways `json:"gateways,omitempty"`
+
+	// Hostnames defines a set of SNI names that should match against the
+	// SNI attribute of TLS ClientHello message in TLS handshake.
+	//
+	// SNI can be "precise" which is a domain name without the terminating
+	// dot of a network host (e.g. "foo.example.com") or "wildcard", which is
+	// a domain name prefixed with a single wildcard label (e.g. `*.example.com`).
+	// The wildcard character `*` must appear by itself as the first DNS label
+	// and matches only a single label. You cannot have a wildcard label by
+	// itself (e.g. Host == `*`).
```

Comment:
Why not? I think right now in the spec the answer would be "Because you should just use TCPRoute instead in that case", but in the future could we not allow routing on other attributes of the TLS handshake?

I don't feel strongly about this *at all* - our implementation only supports routing by SNI today - just curious.

Reply:
I see this is also copy+pasted so definitely not blocking this PR at all